.tekton/pull-request.yaml

```diff
@@ -46,6 +50,30 @@ spec:
             - name: run-tests
               image: registry.access.redhat.com/ubi8/ubi:latest
               script: cd $(workspaces.workspace.path) && ./automation/test.sh
+      - name: run-gitlint
+        params:
+          - name: target-branch
+            value: $(params.target-branch)
+        workspaces:
+          - name: workspace
+            workspace: workspace
+        runAfter:
+          - clone-repository
+        taskSpec:
+          params:
+            - name: target-branch
+          workspaces:
+            - name: workspace
+          steps:
+            - name: run-gitlint
+              image: registry.access.redhat.com/ubi9/python-39:latest
+              workingDir: $(workspaces.workspace.path)
+              script: |
+                #!/bin/bash -ex
+
+                python -m pip install gitlint
+                git fetch --unshallow
+                gitlint --commits "origin/$(params.target-branch)..HEAD"
```

Comment:
Wouldn't this just run glint on the main branch and not on the changes in the PR?

Reply:
It specifics the commits as far as I understand rather than the branch, so it runs on the PR branch in any case as far as I understand, but I think there's another issue here nonetheless:

I think this will require the PR to always be rebased, which is going to be a headache.

Suppose that you had a PR open and in the meantime another PR was merged, then there's no series of commits getting you from the target branch to the tip of your PR branch. I need to check to see what will happen in that case, but I think we need the merge-base stuff to avoid it.

With merge-base, we should be getting an older ancestor as starting point for the check, which should be ok, as the history from that older ancestor  should be compliant with our gitlint anyway. If not, a rebase should address it, but it's still far less rebasing comparing to what's currently here (if I'm right).

Reply:
@yftacherzog Yesterday, in my PR, I was trying to for go for the merge-base approach.
But I ran into an issue:
After the cloning task, I ended up with a detached HEAD of the commit in my PR.
It's fine if you just want to scan that commit but pint requires that you'll be on branch X (with history) and point to the base branch ('main' for example).
I couldn't use merge-base because the detached commit didn't have any history (although the PR commit comes from a common ancestor).

Reply:
@eisraeli I tested a bit more and I think we're good as-is.

Outputting the log of the checked out ref, we do get the proper log of the PR branch. See example here: https://console-openshift-console.apps.stone-prd-rh01.pg1f.p1.openshiftapps.com/k8s/ns/tekton-ci/tekton.dev~v1beta1~PipelineRun/on-pull-request-j7qrc/logs/run-gitlint

Regarding my concern which may require using merge-base, looking at what gitlint does (running with `--debug`), it uses `git rev-list upstream/main..HEAD`  which gives all the commits on HEAD which are not on upstream/main, even if upstream/main progressed and HEAD is not properly rebased.